openedx/features/partners/tests/test_views.py

```diff
@@ -32,6 +33,11 @@ def setUp(self):
         self.registration_url = reverse('partner_register', args=[self.PARTNER])
         self.partner_url = reverse('partner_url', args=[self.PARTNER])
 
+    @classmethod
+    def setUpClass(cls):
+        super(PartnerRegistrationViewTest, cls).setUpClass()
+        configure_philu_theme()
```

Comment:
I believe we need to do this every time our tests have to use philu theme. How about to override this method in `ApiTestCase` something like `PhiluApiTestCase` and inherit our test from it?

Reply:
`ApiTestCase` is not the only base class we use, and there are a lot of test suites already inheriting from `TestCase`. Would like to hear @zahid-arbisoft's POV on this as well.

Reply:
1. We use `configure_philu_theme` to fix issues related to (Philu) templates, and ApiTestCase is used to test APIs. Ideally, an API should not have a template. So do we really need `configure_philu_theme` here?
2. If we are inheriting from `TestCase`, then we can inherit from `PhiluThemeMixin` instead, for views only where we want to set SiteTheme
3. If we are inheriting from `ModuleStoreTestCase`, then we will have to call `configure_philu_theme` and `clear_philu_theme` manually.
4. Always call both `configure_philu_theme` and `clear_philu_theme` in specific way [see detail](https://arbisoft.slack.com/archives/CGYKGH4P5/p1588095021077100)

Here point 1 seems relevant, even if you still want this code than point 4 is a must.

Reply:
@zahid-arbisoft `test_create_new_partner_landing_page_is_accessible` requires for the theme to be configured. This is the test case that was breaking throwing a `TemplateNotFound`.
@ahmed-faraz46 I have used the `PhiluThemeMixin`. This has to be inherited from before the `APITestCase` class. Please have another look at the PR.